zstor/src/actors/zdbfs.rs

```diff
@@ -1,75 +1,146 @@
+    /// Metrics actor address
+    pub metrics: Addr<MetricsActor>,
+    /// ZdbFs stats actor address
+    stats_actor: Addr<ZdbFsStatsActor>,
 }
 
-impl ZdbFsStatsActor {
-    /// Create a new ZdbFsStatsActor, which will get statistics and push them to the metrics actor
-    /// at a given interval.
-    pub fn new(stats: ZdbFsStats, metrics: Addr<MetricsActor>) -> Self {
-        Self { stats, metrics }
+impl ZdbFsStatsProxyActor {
+    /// Create a new zdbfs stats proxy actor
+    pub fn new(buf: PathBuf, metrics: Addr<MetricsActor>) -> Self {
+        let stats_actor = SyncArbiter::start(1, ZdbFsStatsActor::new);
+        ZdbFsStatsProxyActor {
+            buf,
+            metrics,
+            stats_actor,
+        }
     }
-
-    fn get_stats(&mut self, ctx: &mut <Self as Actor>::Context) {
-        ctx.notify(GetStats);
+    fn probe_stats(&mut self, ctx: &mut <Self as Actor>::Context) {
+        ctx.notify(ProbeStatsActor);
     }
 }
 
-/// Message requesting the actor to get the statistics of the 0-db-fs.
-#[derive(Debug, Message)]
+#[derive(Message)]
 #[rtype(result = "()")]
-struct GetStats;
+/// Probes the sync actor to process metrics
+struct ProbeStatsActor;
 
-impl Actor for ZdbFsStatsActor {
+impl Actor for ZdbFsStatsProxyActor {
     type Context = Context<Self>;
 
     fn started(&mut self, ctx: &mut Self::Context) {
         ctx.run_interval(
-            Duration::from_secs(DEFAULT_ZDBFS_STATS_INTERVAL_SECS),
-            Self::get_stats,
+            time::Duration::from_secs(DEFAULT_ZDBFS_STATS_INTERVAL_SECS),
+            Self::probe_stats,
         );
     }
 }
 
-impl Handler<GetStats> for ZdbFsStatsActor {
-    type Result = ResponseActFuture<Self, ()>;
+impl Handler<ProbeStatsActor> for ZdbFsStatsProxyActor {
+    type Result = ResponseFuture<()>;
 
-    fn handle(&mut self, _: GetStats, _: &mut Self::Context) -> Self::Result {
-        let res = self.stats.get_stats();
+    fn handle(&mut self, _: ProbeStatsActor, _: &mut Self::Context) -> Self::Result {
+        let buf = self.buf.clone();
         let metrics = self.metrics.clone();
-        Box::pin(
-            async move {
-                match res {
-                    Err(ref e) => {
-                        warn!("Could not get 0-db-fs stats: {}", e);
-                    }
-                    Ok(stats) => {
-                        if let Err(e) = metrics.send(UpdateZdbFsStats { stats }).await {
-                            error!("Could not update 0-db-fs stats: {}", e);
-                        };
-                    }
-                };
-                res
+        let stats_actor = self.stats_actor.clone();
+        Box::pin(async move {
+            if let Err(e) = stats_actor
+                .send(ProcessStats {
+                    buf: buf.clone(),
+                    metrics: metrics.clone(),
+                })
+                .await
+            {
+                warn!("couldn't process zdbfs stats: {}", e);
+            }
```

Comment:
```suggestion
            if let Err(e) = stats_actor.send(ProcessStats { buf, metrics }).await {
                warn!("couldn't process zdbfs stats: {}", e);
            }
```

Both buf and metrics are already cloned, and are not used again after this call, so these `clone()`'s here are obsolete